lib/gui/modules/analytics.js

```diff
@@ -20,108 +20,62 @@
 const angular = require('angular');
-const username = require('username');
-const isRunningInAsar = require('electron-is-running-in-asar');
-const errors = require('../../shared/errors');
-const os = require('os');
+const _ = require('lodash');
+const resinCorvus = require('resin-corvus/browser');
 const packageJSON = require('../../../package.json');
-const arch = require('arch');
-const utils = require('../../shared/utils');
-
-// Force Mixpanel snippet to load Mixpanel locally
-// instead of using a CDN for performance reasons
-window.MIXPANEL_CUSTOM_LIB_URL = '../../bower_components/mixpanel/mixpanel.js';
 
-require('../../../bower_components/mixpanel/mixpanel-jslib-snippet.js');
-require('../../../bower_components/angular-mixpanel/src/angular-mixpanel');
 const MODULE_NAME = 'Etcher.Modules.Analytics';
 const analytics = angular.module(MODULE_NAME, [
-  'analytics.mixpanel',
   require('../models/settings')
 ]);
 
-/**
- * @summary Get host architecture
- * @function
- * @private
- *
- * @description
- * We need this because node's os.arch() returns the process architecture
- * See: https://github.com/nodejs/node-v0.x-archive/issues/2862
- *
- * @returns {String} Host architecture
- *
- * @example
- * if (getHostArchitecture() === 'x64') {
- *   console.log('Host architecture is x64');
- * }
- */
-const getHostArchitecture = () => {
-  return _.includes([ 'ia32', 'x64' ], process.arch) ? _.replace(arch(), 'x86', 'ia32') : process.arch;
-};
-
-// Mixpanel integration
-// https://github.com/kuhnza/angular-mixpanel
-
-analytics.config(($mixpanelProvider) => {
-  $mixpanelProvider.apiKey('63e5fc4563e00928da67d1226364dd4c');
-
-  $mixpanelProvider.superProperties({
-    electron: process.versions.electron,
-    node: process.version,
-    arch: process.arch,
-    version: packageJSON.version,
-    osPlatform: os.platform(),
-    hostArch: getHostArchitecture(),
-    osRelease: os.release(),
-    cpuCores: os.cpus().length,
-    totalMemory: os.totalmem(),
-    startFreeMemory: os.freemem()
+analytics.config(() => {
+  resinCorvus.install({
+    services: {
+      sentry: _.get(packageJSON, [ 'analytics', 'sentry', 'token' ]),
+      mixpanel: _.get(packageJSON, [ 'analytics', 'mixpanel', 'token' ])
+    },
+    options: {
+      release: packageJSON.version
+    }
   });
 });
 
-// TrackJS integration
-// http://docs.trackjs.com/tracker/framework-integrations
-
-analytics.run(($window) => {
-
-  // Don't configure TrackJS when
-  // running inside the test suite
-  if (window.mocha) {
-    return;
-  }
-
-  $window.trackJs.configure({
-    userId: username.sync(),
-    version: packageJSON.version
-  });
-});
+analytics.service('AnalyticsService', function(SettingsModel) {
 
-analytics.service('AnalyticsService', function($log, $window, $mixpanel, SettingsModel) {
+  /**
+   * @summary Disable or enable logging according to settings
+   * @function
+   * @private
+   *
+   * @example
+   * updateStatus();
+   */
+  const updateStatus = () => {
```

Comment:
I love where this is going. As it is now, it'd be easy for another client to accidentally emit a event/exception forgetting to run this first. As a solution, can `resinCorvus.install` accept a function argument to determine if it should emit things?

For example:

```
resinCorvus.install({
  shouldReport: () => {
    return SettingsModel.get('errorReporting');
  }
});
```

Then, `resinCorvus` will internally call this function whenever needed.